packages/vue-app/template/client.js

```diff
@@ -151,7 +151,7 @@ async function loadAsyncComponents(to, from, next) {
         } else if (Array.isArray(watchQuery)) {
           return watchQuery.some(key => this._diffQuery[key])
         } else if (typeof watchQuery === 'function') {
-          return watchQuery(from.query, to.query)
+          return watchQuery({from: from.query, to: to.query})
```

Comment:
I guess you're adamant that we shouldnt pass the component in one way or another?

Reply:
What do you mean?

Reply:
I think he means not passing route object reference but query. In this case i would suggest keeping consistency with watch API (oldVal, newVal)

Reply:
I meant the same as @pi0 actually about the not-used `this` context. But instead of unresolving the previous thread I created a new one, sorry about that. Will resolve this one in favor of the other

Reply:
😿I’m kind of confused, which watch api do you mean?

Reply:
Oh, you mean vue watch, I got it.